core/node/node_framework/src/implementations/resources/pools.rs

```diff
@@ -1,10 +1,18 @@
+use std::sync::{
+    atomic::{AtomicU32, Ordering},
+    Arc,
+};
+
 use zksync_dal::{connection::ConnectionPoolBuilder, ConnectionPool};
 
 use crate::resource::Resource;
 
 /// Represents a connection pool to the master database.
 #[derive(Debug, Clone)]
-pub struct MasterPoolResource(ConnectionPoolBuilder);
+pub struct MasterPoolResource {
+    connections_count: Arc<AtomicU32>,
```

Comment:
should we set a limit?  Otherwise i don't see much value here

Reply:
I don't think we should. It doesn't change behavior, but it'll make the debugging easier in the future.
Today we don't know how much connections do we use, and we can certainly create more connections than DB supports (which already happened in our CI and caused problems).